src/frontend/Komponenter/Behandling/Brev/BrevUtils.ts

```diff
@@ -34,3 +34,49 @@ export const grupperDelmaler = (delmaler: Delmal[]): { [mappeNavn: string]: Delm
         return acc;
     }, {});
 };
+
+const hentVerdiFraMellomlagerEllerNull = (
+    flettefeltFraMellomlager: FlettefeltMedVerdi[] | undefined,
+    feltId: string
+) => {
+    if (flettefeltFraMellomlager) {
+        return (
+            flettefeltFraMellomlager.find((flettefelt) => flettefelt._ref === feltId)?.verdi || null
+        );
+    }
+    return null;
+};
+export const initFlettefelterMedVerdi = (
+    brevStruktur: BrevStruktur,
+    flettefeltFraMellomlager: FlettefeltMedVerdi[] | undefined
+): FlettefeltMedVerdi[] =>
+    brevStruktur.flettefelter.flettefeltReferanse.map((felt) => ({
+        _ref: felt._id,
+        verdi: hentVerdiFraMellomlagerEllerNull(flettefeltFraMellomlager, felt._id),
+    }));
+export const initValgteFeltMedMellomlager = (
+    valgteFeltFraMellomlager: ValgtFelt | undefined,
+    brevStruktur: BrevStruktur
+): ValgtFelt =>
+    Object.entries(valgteFeltFraMellomlager || {}).reduce(
+        (andreValgteFelt, [valgfeltApiNavn, mulighet]) => {
+            const flettefeltForValgmulighetUtenVerdiFraSanity =
+                brevStruktur.dokument.delmalerSortert.flatMap(
+                    (delmal) =>
+                        delmal.delmalValgfelt
+                            .find((valgFelt) => valgFelt.valgFeltApiNavn === valgfeltApiNavn)
+                            ?.valgMuligheter.find(
+                                (valgmulighet) =>
+                                    valgmulighet.valgmulighet === mulighet.valgmulighet
+                            )?.flettefelter
+                );
+            return {
+                ...andreValgteFelt,
+                [valgfeltApiNavn]: {
+                    ...mulighet,
+                    flettefelter: flettefeltForValgmulighetUtenVerdiFraSanity.filter(Boolean),
+                },
+            };
+        },
+        {}
+    );
```

Comment:
Tar veldig gjerne forslag på hvordan denne kan forenkles. Skal gjøre ca. dette:
- Løp over alle valgteFelt (valgfelt) fra mellomlager
- For hvert valgfelt, finn tilsvarende i brevstruktur (mal fra sanity) og finn alle flettefelt som skal ligge under det valgfeltet/valgmuligheten

Reply:
Jobber litt til med denne